docs/testing/playwright/test-authenticated-flows.mdx

```diff
@@ -0,0 +1,125 @@
+
+  // Configure Playwright with Clerk
+  setup('global setup', async ({}) => {
+    await clerkSetup()
+  })
+
+  // Define the path to the storage file, which is `user.json`
+  const authFile = path.join(__dirname, '../playwright/.clerk/user.json')
+
+  setup('authenticate and save state to storage', async ({ page }) => {
+    // Perform authentication steps.
+    // This example uses a Clerk helper to authenticate
+    await page.goto('/')
+    await clerk.signIn({
+      page,
+      signInParams: {
+        strategy: 'password',
+        identifier: process.env.E2E_CLERK_USER_USERNAME!,
+        password: process.env.E2E_CLERK_USER_PASSWORD!,
+      },
+    })
+    await page.goto('/protected')
+    // Ensure the user has successfully accessed the protected page
+    // by checking an element on the page that only the authenticated user can access
+    await page.waitForSelector("h1:has-text('This is a PROTECTED page')")
+
+    await page.context().storageState({ path: authFile })
+  })
+  ```
+
+  ### Load the stored auth state in your tests
+
+  You can either load the stored auth state [in the config](#-in-the-config) or directly [in a test file](#-in-a-test-file). Loading in the config is useful if you want to authenticate once and reuse the same auth state for all tests or groups of tests. Loading in a test file is useful if you want to authenticate for a specific test case.
+
+  #### In the config
+
+  In your `playwright.config.ts`, create a `global setup` project and declare it as a [dependency](https://playwright.dev/docs/test-projects#dependencies) for all your testing projects. This means that the `global setup` project will always run before all the tests, and because it's where you prepared auth state, it will authenticate before all the tests. All testing projects should use the authenticated state as `storageState`.
+
+  ```tsx {{ filename: 'playwright.config.ts' }}
+  // ...
+  projects: [
+    {
+      name: 'global setup',
+      testMatch: /global\.setup\.ts/,
+    },
+    {
+      name: 'Main tests',
+      testMatch: /.*app.spec.ts/,
+      use: {
+        ...devices['Desktop Chrome'],
+      },
+      dependencies: ['global setup'],
+    },
+    {
+      name: 'Authenticated tests',
+      testMatch: /.*authenticated.spec.ts/,
+      use: {
+        ...devices['Desktop Chrome'],
+
+        // Use prepared Clerk auth state
+        storageState: 'playwright/.clerk/user.json',
+      },
+      dependencies: ['global setup'],
+    },
+  ]
+  ```
+
+  #### In a test file
+
+  To use the stored auth state in a test file, see the following example:
+
+  ```tsx {{ filename: 'authenticated.spec.ts', mark: [4, 5] }}
+  import { setupClerkTestingToken } from '@clerk/testing/playwright'
+  import { test } from '@playwright/test'
+
+  // Use prepared Clerk auth state
+  test.use({ storageState: 'playwright/.clerk/user.json' })
+
+  test('user test', async ({ page }) => {
+    await setupClerkTestingToken({ page })
```

Comment:
I added this `await setupClerkTestingToken` to the example because I thought its required for tests - is this accurate?

Reply:
Actually, it's not needed in this case.

Currently, the Testing Tokens are used only for sign-ups, but in the future we may also use them on sign-in.

In this scenario, you are already using an active session, so you don't really need that.